From b1a76fca2ad454ab5688448731181235b23cfaba Mon Sep 17 00:00:00 2001
From: Vincent Bernat <vincent@bernat.ch>
Date: Tue, 13 Oct 2026 23:07:45 +0200
Subject: [PATCH] Allow null SRV records (.) in hostname validator trailing dot
 check

This is similar to what was done for MX records (see #74). This is
allowed in RFC 2782:

> A Target of "." means that the service is decidedly not available at
> this domain.
---
diff --git a/internal/dnsrecordresourcevalidator/hostname.go b/internal/dnsrecordresourcevalidator/hostname.go
index 6b23b2d19f19..8411a687e4b5 100644
--- a/internal/dnsrecordresourcevalidator/hostname.go
+++ b/internal/dnsrecordresourcevalidator/hostname.go
@@ -47,9 +47,9 @@ func (v hostnameValidator) ValidateResource(ctx context.Context, req resource.Va
 		return
 	}
 
-	isNullMXRecord := rType == "MX" && value == "."
+	isNullRecord := (rType == "MX" || rType == "SRV") && value == "."
 	valueIsHostname := rType == "CNAME" || rType == "MX" || rType == "NS" || rType == "PTR" || rType == "SRV"
-	if valueIsHostname && !isNullMXRecord && value[len(value)-1] == '.' {
+	if valueIsHostname && !isNullRecord && value[len(value)-1] == '.' {
 		resp.Diagnostics.Append(diag.NewAttributeErrorDiagnostic(valueAttr, "Invalid attribute configuration", v.Description(ctx)))
 	}
 }
